src/util/irep_serialization.cpp

```diff
@@ -149,13 +146,11 @@ std::size_t irep_serializationt::insert_on_read(
     ireps_container.ireps_on_read.resize(1+id*2,
       std::pair<bool, irept>(false, get_nil_irep()));
 
-  if(ireps_container.ireps_on_read[id].first)
-    throw "irep id read twice.";
-  else
-  {
-    ireps_container.ireps_on_read[id]=
-      std::pair<bool, irept>(true, i);
-  }
+  DATA_INVARIANT(ireps_container.ireps_on_read[id].first==false,
```

Comment:
I'm not sure about this one, this _might_ be an exception? Could also be a proper `DATA_INVARIANT`, would require a closer look